FacebookLoginProvider.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  FacebookLoginProvider.swift
+//  Authentication
+//
+//  Created by Gustavo Cairo on 1/13/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import Foundation
+import ReactiveCocoa
+import ReactiveSwift
+import FacebookLogin
+import FacebookCore
+import enum Result.NoError
+
+/**
+    Simple struct to encapsulate a Facebook profile request.
+ */
+internal struct FacebookProfileRequest: GraphRequestProtocol {
+    struct Response: GraphResponseProtocol {
+        var email: String?
+        var name: String?
+        var userId: Int64?
+        
+        init(rawResponse: Any?) {
+            print(rawResponse!)
+            if let dict = rawResponse as? [String: Any] {
+                email = (dict["email"] as? String)!
+                name = (dict["name"] as? String)!
+                userId = Int64((dict["id"] as? String)!)
+            }
+        }
+    }
+    
+    var graphPath = "/me"
+    var parameters: [String : Any]? = ["fields": "name, id, email"]
+    var accessToken = AccessToken.current
+    var httpMethod: GraphRequestHTTPMethod = .GET
+    var apiVersion: GraphAPIVersion = .defaultVersion
+}
+
+/**
+    User returned after logging in with Facebook.
+ 
+    Except for the access token, all other properties are optional
+    because they are requested in a separate request after the login
+    is completed. If this request fails, the only information the user
+    will contain is its corresponding AccessToken.
```

Comment:
I would add some more documentation of what they can do with it.
Like for example "The AccessToken is the only requirement for requesting more information about the user from Facebook. You should use the FacebookCore framework for these requests." (or something like that)